### PR TITLE
Pieces controller

### PR DESCRIPTION
**Pull request by mahalocode**
### What does this PR do?
[Support moving pieces. Later we can make the pieces draggable/droppable and remove the interstitial page.]

### Where should I start?
[pieces_controller.rb/view pages]

### How do I manually test this?
[Startup local directory. Login and join a game highlight each piece to see coordinates. Click the on any piece and check URL.]

### Additional Comments
[The chess board looks off right now. Is not a major problem.]

### Trello story
[List or link]

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ def index
   end
 
   def show
+    @pieces = Piece.find(params[:piece_id])
```
Comment (danielpowell4) [inline app/controllers/games_controller.rb]: Why are you assigning this twice?

File: app/assets/javascripts/pieces.js.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (danielpowell4) [inline app/assets/javascripts/pieces.js.coffee]: No need to commit these

File: app/controllers/games_controller.rb [hunk 3]
```diff
@@ -35,6 +36,11 @@ def destroy
     redirect_to games_path, notice: 'Your game has been cancel'
   end
 
+  def move_to!
```
Comment (danielpowell4) [inline app/controllers/games_controller.rb]: I think this is included by mistake. 

Duplicate code found in pieces controller. The piece_params you try to pass into `@piece.update_attributes` isn't accessible here. In a private method in the other controller...

File: app/controllers/pieces_controller.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    @pieces = @game.pieces
+    @piece_id = params[:piece_id]
+    @x_coord = params[:x_coord]
+    @y_coord = params[:y_coord]
+  end
+
+  def update
+  	@pieces = Piece.find(params[:piece_id])
+  	@game = @pieces.game
+  	@pieces.update_attributes piece_params
```
Comment (danielpowell4) [inline app/controllers/pieces_controller.rb]: Should `valid_move?` and `move_to!` be called before up `update_attributes`? I think yes, but am unfamiliar with your strategy.

File: config/initializers/devise.rb [hunk 5]
```diff
@@ -13,7 +13,7 @@
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
   config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
-
+  config.secret_key = '2c663a3df976982204d685e467ddb224c8fc44e67820567bbabd66e945110afbe47b6e608486ac7ed7f8414d0e878d780d6749337075580e52af337222a1b081'
```
Comment (danielpowell4) [inline config/initializers/devise.rb]: This isn't a good place for a secret key.

Should be in Figaro and Heroku, not version control for all to see

File: spec/helpers/pieces_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (danielpowell4) [inline spec/helpers/pieces_helper_spec.rb]: Auto generated file. Delete as appropriate (at least remove pending test)

File: spec/controllers/pieces_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (danielpowell4) [inline spec/controllers/pieces_controller_spec.rb]: Technically there should be tests here (or drop the file?)

**Comment by mahalocode** [comment]
Thank you. I'm not at my desktop right now but will fix it asap.

**Comment by danielpowell4** [comment]
Not a huge hurry. I'm looking into the broken show page now. Should be able to fix somewhat easily. I'm not sure that you will need everything to be a link in the nearish future.

So I don't want to invest too much time into it

**Comment by icrustandi** [comment]
Is this still waiting on more changes? Right now the show page looks off 
image

**Comment by danielpowell4** [comment]
Merging this in and will fix as appropriate locally so this is no longer a blocker

**Pull request merged**